src/modules/dream/readout/DataParser.cpp

```diff
@@ -29,67 +28,59 @@ int DataParser::parse(const char *Buffer, unsigned int Size) {
   while (BytesLeft) {
     // Parse Data Header
     if (BytesLeft < sizeof(ESSReadout::Parser::DataHeader)) {
-      XTRACE(DATA, DEB, "Not enough data left for header: %u", BytesLeft);
-      Stats.ErrorHeaders++;
+      XTRACE(DATA, WAR, "Not enough data left for header: %u", BytesLeft);
+      Stats.ErrorDataHeaders++;
       Stats.ErrorBytes += BytesLeft;
       return ParsedReadouts;
     }
 
-    auto DataHdrPtr = (ESSReadout::Parser::DataHeader *)DataPtr;
+    auto Data = (DreamReadout *)((char *)DataPtr);
 
-    if (BytesLeft < DataHdrPtr->DataLength) {
-      XTRACE(DATA, DEB, "Data size mismatch, header says %u got %d",
-             DataHdrPtr->DataLength, BytesLeft);
-      Stats.ErrorHeaders++;
+    ///\todo clarify distinction between logical and physical rings
+    // for now just divide by two
+    Data->RingId = Data->RingId / 2;
```

Comment:
We need to decide where this happens - some modules /2 in the data parser, others in the instrument

Reply:
Agree. I would say as early as possible in general.